src/project/util/projectUtils.ts

```diff
@@ -113,9 +112,7 @@ export function doWithFiles(project: ProjectNonBlocking,
                 .on("end", _ => {
                     resolve(Promise.all(filePromises));
                 });
-        });
-    };
-    return runOrDefer<ProjectScripting, File[]>(project, funrun);
+        }).then(files => project);
```

Comment:
wait does this imply that it already does return the Project? ... then maybe that other place doesn't need to do the .then(files=>project) since it's here